api/secrets/initialize.js

```diff
@@ -247,62 +255,238 @@ function _initializeVault(bag, next) {
   );
 }
 
+function _checkInitStatus(bag, next) {
+  var who = bag.who + '|' + _checkInitStatus.name;
+  logger.verbose(who, 'Inside');
+
+  var client = new VaultAdapter(bag.vaultUrl);
+
+  client.getInitializedStatus(
+    function (err, response) {
+      if (err)
+        return next(
+          new ActErr(who, ActErr.OperationFailed,
+            'Failed to get vault initialized status: ' + util.inspect(err))
+        );
+
+      if (response.initialized === true) {
+        logger.debug('Vault already initialized');
+
+        if (_.isEmpty(bag.config.unsealKey1) ||
+          _.isEmpty(bag.config.unsealKey2) ||
+            _.isEmpty(bag.config.unsealKey3) ||
+              _.isEmpty(bag.config.unsealKey4) ||
+                _.isEmpty(bag.config.unsealKey5) ||
+                  _.isEmpty(bag.config.rootToken)) {
+
+          return next(
+            new ActErr(who, ActErr.OperationFailed,
+              'Vault needs 5 unseal keys. One(or more) unseal keys are ' +
+              ' missing. Reset vault server or make sure keys are present')
+          );
+        } else {
```

Comment:
You don't need the `else` because the `if` has `return`.  But, it might be easier to read if you just set `bag.config.isInitialized` to `response.initialized`.